extensions/src/objectDetection/index.ts

```diff
@@ -0,0 +1,161 @@
+   * The thickness of the detection box
+   */
+  thickness: number;
+
+  /**
+   * Initializes the extension with standard values for the extension's class attributes.
+   */
+  async init() {
+    this.enableVideo()
+    this.continuous = false;
+    this.detector = await initializeObjectDetector();
+    this.imageHelper = getImageHelper(this.DIMENSIONS[0], this.DIMENSIONS[1])
+    this.color = 'white'
+    this.thickness = 5
+    this.processFreq = 100;
+  }
+
+  /**
+   * Runs a loop for constantly updating the video frame with a new set of detections.
+   * Used for the continuous detection block.
+   */
+  private async detectionLoop() {
+    while (this.continuous) {
+      const frame = this.getVideoFrame("canvas");
+      const start = Date.now();
+      if (frame) {
+        const detections = await this.detector.detect(frame);
+        this.displayImageDetections(detections);
+      }
+      const elapsed = Date.now() - start;
+      await untilTimePassed(this.processFreq - elapsed);
+      this.clearFrame()
+    }
+  }
+
+  /**
+   * Clears the frame of all drawables (resets the frame to blank video).
+   */
+  clearFrame() {
+    while (this.drawables.length > 0) this.drawables.shift().destroy();
+  }
+
+  /**
+   * Displays the detections from the detector in the video frame.
+   * @param detections The detections object returned from the detector's detect method
+   */
+  async displayImageDetections(detections) {
+    const { drawables, imageHelper } = this;
+    const rects = imageHelper.createRects(detections.detections, this.color, this.thickness)
+    drawables.push(this.createDrawable(rects));
+  }
+
+  @block({
+    type: "command",
+    text: (delay) => `Set detection rate to ${delay}`,
+    arg: {
+      type: "number",
+      options: [60, 30, 10, 2, 1],
+      defaultValue: 10
+    } as const
+  })
+  setFrameRate(delay: number) {
+    this.processFreq = 1000 / delay
+  }
+
+  @block({
+    type: "command",
+    text: (color) => `Set box color to ${color}`,
+    arg: "color"
+  })
+  setColor(color: RGBObject) {
+    this.color = rgbToHex(color);
+  }
+
+  @block({
+    type: "command",
+    text: (thickness) => `Set box thickness to ${thickness}`,
+    arg: {
+      type: "number",
+      defaultValue: 5,
+      handler: (x) => {
+        return Math.min(Math.max(0, x), 20)
+      }
+    }
+  })
+  setThickness(thickness: number) {
+    this.thickness = thickness;
+  }
+
+  @block({
+    type: BlockType.Command,
+    text: (freezeTime) => `Detect objects for ${freezeTime} seconds`,
+    arg: { type: ArgumentType.Number, defaultValue: 1 }
+  })
+  async detectObject(freezeTime: number) {
+    const frame = this.getVideoFrame('canvas')
+    const detections = await this.detector.detect(frame);
+    this.displayImageDetections(detections);
+    await untilTimePassed(freezeTime * 1000)
+    this.clearFrame()
+  }
+
+  @block({
+    type: BlockType.Command,
+    text: (state) => `Toggle continuous detection ${state}`,
```

Comment:
I think "toggle" might be a confusing word here. Should we just say "Turn continuous detection on/off"??